packages/plugins/README.md

```diff
@@ -180,19 +180,19 @@ _Parameters_
 
 _Returns_
 
--   `WPPlugin | undefined`: The previous plugin settings object, if it has been successfully unregistered; otherwise `undefined`.
+-   The previous plugin settings object, if it has been successfully unregistered; otherwise `undefined`.
 
 #### withPluginContext
 
 A Higher Order Component used to inject Plugin context to the wrapped component.
 
 _Parameters_
 
--   _mapContextToProps_ `Function`: Function called on every context change, expected to return object of props to merge with the component's own props.
+-   _mapContextToProps_ `( context: PluginContext, props: T ) => T & PluginContext`: Function called on every context change, expected to return object of props to merge with the component's own props.
 
 _Returns_
 
--   `WPComponent`: Enhanced component with injected context as props.
```

Comment:
This one is harder to fix, because none of these types (`WPElement`, `WPComponent`, `JSX.Element`, `React.ComponentType`, `React.Element`) are compatible with what typescript infers from `createHigherOrderComponent`. So it's a lot easier to just let inference do its magic here. (I'm also not convinced that having the "WPComponent" text in the docs really provides enough meaningful value to try to fix this issue :/)

Reply:
I take that back; apparently, we can just put WPComponent in JSdoc, and it's happy enough 😅